tools/github/scripts/Sync-AzsdkLabels.ps1

```diff
@@ -42,41 +46,75 @@ if ($PSCmdlet.ParameterSetName -eq 'RepositoryFile') {
     $Repositories = Get-Content $RepositoryFilePath
 }
 
-# Filter out the source repository.
-$Repositories = $Repositories.Where({$_ -ne $SourceRepository})
+# Extract the label content to be sync'd.
+$labels = Import-Csv $LabelsFilePath -Header "Name", "Description", "Color"
+
+# Synchronize labels for each repository.
+$activity = "Synchronizing labels"
+Write-Progress -Activity $activity -PercentComplete 0
+
+$totalLabels = $labels.Count * $Repositories.Count
+$labelCount = 1
 
 foreach ($repo in $Repositories) {
     if ($Force -or $PSCmdlet.ShouldProcess(
-        "Cloning labels from $SourceRepository to $repo",
-        "Clone labels from $SourceRepository to $repo?",
-        "Clone labels")) {
-        $result = gh -R "$repo" label clone "$SourceRepository" --force 2>&1
-        if ($LASTEXITCODE) {
-            Write-Error "Failed to clone labels from $SourceRepository to ${repo}: $result"
+        "Writing common labels to $repo",
+        "Write common labels to $repo",
+        "Create or update common labels")) {
+
+        foreach ($label in $labels) {
+            $result = gh label create $label.Name -d $label.Description -c $label.Color -R "$repo" --force 2>&1
+
+            if ($LASTEXITCODE) {
+                Write-Error "Failed to create or update the common labels to ${repo}: $result"
+            }
+
+            $completed = ($labelCount / $totalLabels) * 100
+            Write-Progress -Activity $activity -Status "$($repo): $($label.Name)" -PercentComplete $completed
+            $labelCount++
```

Comment:
Wrong order. You already completed the label. Start with 0 and increment, then write progress. This would be +1 off when done. Probably not noticeable, but still wrong.